scripts/pharmgkb/drug_gene_relations/conversion/generate_conversion_files.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates the xref to dcid conversion files
+
+Warning: This file takes a very long time to run (at least 3 hours).
+
+(PharmGKB --> Chembl) and (PubChem Compound ID --> Chembl) both use UniChem API.
+For more info: https://www.ebi.ac.uk/unichem/info/webservices#GetSrcCpdIds
+
+(InChI --> InChI Key) uses InChI API. For more info:
+https://www.chemspider.com/InChI.asmx?op=InChIToInChIKey
+
+(InChI Key --> Chembl) uses chembl Python API. For more info:
+https://github.com/chembl/chembl_webresource_client
+
+Requirements:
+    pip install chembl_webresource_client
+"""
+import sys
+from xml.etree import ElementTree
+import requests
+import pandas as pd
+from chembl_webresource_client.new_client import new_client
+
+
+def pharm_to_chembl(pharm_ids):
+    """Return list of chembl ids that positionally map to pharm ids."""
+    # pharmgkb accession id to chembl id via unichem
+    chembls = []
+    for pharm_id in pharm_ids:
```

Comment:
For long loops like these, it is often useful to print out progress every N items, like

```
  ndone = 0
  for ...:
     ...
     ndone += 1
     if ndone % 10 == 0:
       print('... completed ' + ndone + ' / ' + len(pharm_ids))
```